NDTensors/src/lib/UnallocatedArrays/src/abstractunallocatedarray.jl

```diff
@@ -0,0 +1,65 @@
+using FillArrays: FillArrays, getindex_value
+using NDTensors.SetParameters: set_parameters
+using Adapt: adapt
+
+const UnallocatedArray{ElT,N,AxesT,AllocT} = Union{
+  UnallocatedFill{ElT,N,AxesT,AllocT},UnallocatedZeros{ElT,N,AxesT,AllocT}
+}
+
+@inline Base.axes(A::UnallocatedArray) = axes(parent(A))
+Base.size(A::UnallocatedArray) = size(parent(A))
+function FillArrays.getindex_value(A::UnallocatedArray)
+  return getindex_value(parent(A))
+end
+
+function Base.complex(A::UnallocatedArray)
+  return complex(eltype(A)).(A)
+end
+
+function Base.transpose(a::UnallocatedArray)
+  return set_alloctype(transpose(parent(a)), alloctype(a))
+end
+
+function Base.adjoint(a::UnallocatedArray)
+  return set_alloctype(adjoint(parent(a)), alloctype(a))
+end
+
+function set_alloctype(T::Type{<:UnallocatedArray}, alloc::Type{<:AbstractArray})
+  return set_parameters(T, Position{4}(), alloc)
+end
+
+## This overloads the definition defined in `FillArrays.jl`
+for STYPE in (:AbstractArray, :AbstractFill)
+  @eval begin
+    @inline $STYPE{T}(F::UnallocatedArray{T}) where {T} = F
+    @inline $STYPE{T,N}(F::UnallocatedArray{T,N}) where {T,N} = F
+  end
+end
+
+function allocate(f::UnallocatedArray)
+  a = similar(f)
+  fill!(a, getindex_value(f))
+  return a
+end
+
+function allocate(arraytype::Type{<:AbstractArray}, elt::Type, axes::Tuple)
```

Comment:
```suggestion
function allocate(arraytype::Type{<:AbstractArray}, elt::Type, axes)
```
I would just leave that off and let `similar` handle whatever types it can handle.

Reply:
I removed this from the allocate call and all works with similar as its defined per my changes in the comment above.